luxonis_ml/data/loaders/luxonis_loader.py

```diff
@@ -161,11 +174,12 @@ def _load_image_with_annotations(self, idx: int) -> Tuple[np.ndarray, Labels]:
             present annotations
         """
 
-        uuid = self.instances[idx]
-        df = self.df.filter(pl.col("uuid") == uuid)
+        ann_indices = self.idx_to_df_row[idx]
+        ann_rows = [self.df.row(row) for row in ann_indices]
         if not self.dataset.is_remote:
-            img_path = list(df.select("original_filepath"))[0][0]
+            img_path = ann_rows[0][8]
```

Comment:
Minor comment, but I believe this is a tuple and it might be easier to read if we use some other data type like a dict, so we have some keys and more well-defined structure in the `ann_rows`. I'm not sure how likely it is, but in the case that the tuple element order changes, it would break everything.

Reply:
There is an option to add `named=True` to `self.df.row()`, which produces a dictionary instead of a tuple, but at some performance cost. IMHO having it as a tuple is okay, if we'd change the order of elements in the parquet file or anything, it will get caught in tests.